File: diamond_test.rb
require 'minitest/autorun'
require_relative 'diamond'

describe 'diamond' do
  it "affiche A" do
    valeur_entree = 'A'
    valeur_attendue = 'A'
    resultat = creer_diamant(valeur_entree)
    assert_equal(valeur_attendue, resultat)
  end

  it 'affiche le diamant de B' do
    valeur_entree = 'B'
    valeur_attendue = " A \nB B\n A "

    resultat = creer_diamant(valeur_entree)
    assert_equal(valeur_attendue, resultat)
  end

  it 'affiche le diamant de C' do
    valeur_entree = 'C'
    valeur_attendue = "  A  \n B B \nC   C\n B B \n  A  "

    resultat = creer_diamant(valeur_entree)
    assert_equal(valeur_attendue, resultat)
  end

end

File: test/diamond_test.js
var genereDiamant = require('../diamond.js')
var assert = require('assert');

describe('diamond', function() {
  it('affiche A', function() {
    var lettre = 'A';
    var diamant_attendu = 'A';

    var resultat = genereDiamant(lettre);

    assert.equal(diamant_attendu, resultat);
  });


  it('affiche le diamant de B', function() {
    var lettre = 'B';
    var diamant_attendu = " A \nB B\n A ";

    var resultat = genereDiamant(lettre);

    assert.equal(diamant_attendu, resultat);
  });

  it('affiche le diamant de C', function() {
    var lettre = 'C';
    var diamant_attendu = "  A  \n B B \nC   C\n B B \n  A  ";

    var resultat = genereDiamant(lettre);

    assert.equal(diamant_attendu, resultat);
  });
});
